### PR TITLE
[Java] [DateTime] Port BaseTimeExtractor from C# to Java

### PR DESCRIPTION
**Pull request by VictorGrycuk**
## Modifications
The following files were ported from C# to Java:
- EnglishTimeExtractorConfiguration.java
- BaseTimeExtractor.java
- DateTimeExtractorTest.java
## Testing
All tests are passing
tests

### CODE REVIEW

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/english/extractors/EnglishTimeExtractorConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    // part 3: regex for time
+    // --------------------------------------
+    // handle "at four" "at 3"
+    public static final Pattern AtRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.AtRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern IshRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.IshRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern TimeUnitRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.TimeUnitRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern ConnectNumRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.ConnectNumRegex, Pattern.CASE_INSENSITIVE);
+    public static final Pattern TimeBeforeAfterRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.TimeBeforeAfterRegex, Pattern.CASE_INSENSITIVE);
+
+    public static final Iterable<Pattern> TimeRegexList = new ArrayList<Pattern>()
```
Comment (joseques) [inline Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/english/extractors/EnglishTimeExtractorConfiguration.java]: NIT: Code standard. Curly brace should be on the same line as the declaration. 👮

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/english/extractors/EnglishTimeExtractorConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+        // (three min past)? 3h00 (pm)?
+        add(RegExpUtility.getSafeRegExp(EnglishDateTime.TimeRegex10, Pattern.CASE_INSENSITIVE));
+
+        // 340pm
+        add(ConnectNumRegex);}
+    };
+
+    public final Iterable<Pattern> getTimeRegexList()
+    {
```
Comment (joseques) [inline Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/english/extractors/EnglishTimeExtractorConfiguration.java]: NIT: Code standard. Curly brace should be on the same line as the declaration. 👮

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/english/extractors/EnglishTimeExtractorConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    public EnglishTimeExtractorConfiguration()
+    {
+        this(DateTimeOptions.None);
+    }
+
+    //C# TO JAVA CONVERTER NOTE: Java does not support optional parameters. Overloaded method(s) are created above:
+    //ORIGINAL LINE: public EnglishTimeExtractorConfiguration(DateTimeOptions options = DateTimeOptions.None)
+    public EnglishTimeExtractorConfiguration(DateTimeOptions options)
+    {
```
Comment (joseques) [inline Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/english/extractors/EnglishTimeExtractorConfiguration.java]: NIT: Code standard. Curly brace should be on the same line as the declaration. 👮 (And all the above get* methods)

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseTimeExtractor.java [hunk 4]
```diff
@@ -19,14 +28,120 @@ public String getExtractorName() {
...
+        return timeErs;
     }
 
     @Override
     public List<ExtractResult> extract(String input) {
         return this.extract(input, LocalDateTime.now());
     }
+
+    private List<ExtractResult> MergeTimeZones(List<ExtractResult> timeErs, List<ExtractResult> timeZoneErs, String text)
+    {
```
Comment (joseques) [inline Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseTimeExtractor.java]: NIT: Code standard. Curly brace should be on the same line as the declaration. 👮

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseTimeExtractor.java [hunk 5]
```diff
@@ -19,14 +28,120 @@ public String getExtractorName() {
...
+                {
+                    ret.add(new Token(match.index, match.index + match.length));
+                }
+            }
+        }
+        return ret;
+    }
+
+    private List<Token> SpecialCasesRegexMatch (String text)
+    {
```
Comment (joseques) [inline Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseTimeExtractor.java]: NIT: Code standard. Curly brace should be on the same line as the declaration. 👮 (And all the above functions and methods)

**Review by joseques — CHANGES_REQUESTED** [review]
Minor changes. Everything else :shipit:

**Comment by joseques** [comment]
Good to go :shipit:

**Comment by JuanAr** [comment]
We need to merge this one and since feedback was tackled I'll merge it.

**Review by joseques — APPROVED** [review]

**Pull request merged**